src/gcodeExport.cpp

```diff
@@ -683,7 +697,15 @@ void GCodeExport::writeFXYZE(double speed, int x, int y, int z, double e, PrintF
     }
     if (e != current_e_value)
     {
-        *output_stream << " " << extruder_attr[current_extruder].extruderCharacter << PrecisionedDouble{5, e};
+        *output_stream << " " << extruder_attr[current_extruder].extruderCharacter;
+        if (relative_extrusion)
```

Comment:
I think the code will be a lot more concise when you put it like this:
```
double output_e = (relative_extrusion)? e - current_e_value : e;
*output_stream << " " << extruder_attr[current_extruder].extruderCharacter << PrecisionedDouble{5, output_e};
```

Same applies for the other code changes below.